packages/@react-aria/utils/src/platform.ts

```diff
@@ -11,27 +11,32 @@
  */
 
 function testUserAgent(re: RegExp) {
-  return typeof window !== 'undefined' && window.navigator != null
-    ? re.test(window.navigator.userAgent)
-    : false;
+  if (typeof window === 'undefined' || window.navigator == null) {
+    return false;
+  }
+  return (
+    window.navigator['userAgentData'] &&
+    window.navigator['userAgentData'].brands.some((brand: {brand: string, version: string}) => re.test(brand.brand))
+  ) ||
+  re.test(window.navigator.userAgent);
```

Comment:
First see if [`window.navigator.userAgentData`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgentData) is available, and if so, test to see if one or more of the `window.navigator.userAgentData.brands` matches the regexp, otherwise fallback to test the regexp against `window.navigator.userAgent`.

Reply:
Note, I use `window.navigator['userAgentData']`, because `userAgentData` is not currently included as a [Navigator property in TypeScript](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.navigator.html).